### PR TITLE
doc: readme for overall project (#2)

### LINKED ISSUE #2: README.md
* Fancy badges (optional)
* Project description
* Setup & Installation
* File tree & description of main files
* Basic Workflow & Regulations
  * Link to issues
  * Commit message ([ref](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit))
  * Pull request (at least 2 approval), keep all commits (not squash)
* Statement of contribution...

### PR DESCRIPTION
**Pull request by nichujie**

### CODE REVIEW

**Review by spelgubbe — APPROVED** [review]
Looks good to me!

File: README.md [hunk 1]
```diff
@@ -5,18 +5,113 @@ interceptor should be launched or not. The decision is based on radar tracking
...
-documentation as a markdown file in the repo, and one leader.
-We try to have one commit for each pull request and have one required reviewer.
-Also, issues should be linked to in commit messages.
-When fixing code after a review comment on a pull request,
-fixup commits should be made. If the branch that is being merged to main is
-behind main, git rebase is used and not git merge. Right now,
-we are investigating and working with these tools and principles.
+## Requirement
+
+* JDK 1.7 or above
```
Comment (mdinler) [inline README.md]: 1.7 is Java 7. You can change it to 17. Our Maven config compiles Java 17 code.

File: README.md [hunk 2]
```diff
@@ -5,18 +5,113 @@ interceptor should be launched or not. The decision is based on radar tracking
...
+* Apache Maven 3.3+
+
+## Build & Test
+
+The project is designed as a module for a larger project, thus it has not main functions to execute directly.
+
+You can build the package using:
+```bash
+mvn package
+```
```
Comment (mdinler) [inline README.md]: `mvn test` includes `mvn package`. I would suggest to remove this part.

**Review by mdinler — COMMENTED** [review]
It looks nice, good job. Just don't forget to change 1.7 to 17 🙂

**Pull request merged**